tests/ijby/test-cut.py

```diff
@@ -0,0 +1,213 @@
+    msg = "Expected a boolean, instead got <class 'int'>"
+    DT = dt.Frame(range(10))
+    with pytest.raises(TypeError, match=msg):
+        cut(DT, right_closed = 1492)
+
+
+def test_cut_error_groupby():
+    msg = r"cut\(\) cannot be used in a groupby context"
+    DT = dt.Frame(range(10))
+    with pytest.raises(NotImplementedError, match=msg):
+        DT[:, cut(f[0]), f[0]]
+
+
+def test_cut_empty_frame():
+	DT_cut = cut(dt.Frame())
+	assert_equals(DT_cut, dt.Frame())
+
+
+def test_cut_zero_bins():
+	DT_cut = cut(dt.Frame(range(10)), 0)
+	assert_equals(DT_cut, dt.Frame([None] * 10, stypes = [stype.int32]))
+
+
+def test_cut_trivial():
+	DT = dt.Frame({"trivial": range(10)})
+	DT_cut = cut(DT)
+	assert_equals(DT, DT_cut)
+
+
+def test_cut_expr():
+	DT = dt.Frame([range(0, 30, 3), range(0, 20, 2)])
+	DT_cut = DT[:, cut(cut(f[0] - f[1]))]
+	assert_equals(dt.Frame(range(10)), DT_cut)
+
+
+def test_cut_one_row():
+	DT = dt.Frame([[True], [404], [3.1415926], [None]])
+	DT_cut_right = cut(DT)
+	DT_cut_left = cut(DT, right_closed = False)
+	assert(DT_cut_right.to_list() == [[4], [4], [4], [None]])
+	assert(DT_cut_left.to_list() == [[5], [5], [5], [None]])
+
+
+def test_cut_small():
+	nbins = [4, 2, 5, 4, 10, 3, 2, 5]
+	colnames = ["bool", "int_pos", "int_neg", "int", "float",
+	            "inf_max", "inf_min", "inf"]
+
+	DT = dt.Frame(
+	       [[True, None, False, False, True, None],
+	       [3, None, 4, 1, 5, 4],
+	       [-5, -1, -1, -1, None, 0],
+	       [None, -5, -314, 0, 5, 314],
+	       [None, 1.4, 4.1, 1.5, 5.9, 1.4],
+	       [math.inf, 1.4, 4.1, 1.5, 5.9, 1.4],
+	       [-math.inf, 1.4, 4.1, 1.5, 5.9, 1.4],
+	       [-math.inf, 1.4, 4.1, math.inf, 5.9, 1.4]],
+	       names = colnames
+	     )
+	DT_ref_right = dt.Frame(
+			         [[3, None, 0, 0, 3, None],
+			         [0, None, 1, 0, 1, 1],
+			         [0, 3, 3, 3, None, 4],
+			         [None, 1, 0, 1, 2, 3],
+			         [None, 0, 5, 0, 9, 0],
+			         [None] * DT.nrows,
+			         [None] * DT.nrows,
+			         [None] * DT.nrows],
+			         names = colnames,
+			         stypes = [stype.int32] * DT.ncols
+	               )
+
+	DT_ref_left = dt.Frame(
+			        [[3, None, 0, 0, 3, None],
+			        [1, None, 1, 0, 1, 1],
+			        [0, 4, 4, 4, None, 4],
+			        [None, 1, 0, 2, 2, 3],
+			        [None, 0, 6, 0, 9, 0],
+			        [None] * DT.nrows,
+			        [None] * DT.nrows,
+			        [None] * DT.nrows],
+			        names = colnames,
+			        stypes = [stype.int32] * DT.ncols
+	              )
+
+	DT_cut_list = cut(DT, nbins = nbins)
+	DT_cut_tuple = cut(DT, nbins = tuple(nbins))
+	DT_cut_list_left = cut(DT, nbins = nbins, right_closed = False)
+	assert_equals(DT_ref_right, DT_cut_list)
+	assert_equals(DT_ref_right, DT_cut_tuple)
+	assert_equals(DT_ref_left, DT_cut_list_left)
+
+
+
+@pytest.mark.xfail(reason="This test may fail rarely due to pandas "
+                   "inconsistency, see test_cut_pandas_issue_35126 test")
```

Comment:
Usually we mark `xfail` test that is known to fail right now, but there is an expectation that the functionality will be fixed in the future. In this case, is there an expectation that this test will ever work? I mean, did pandas acknowledge they are going to fix their issue?

Reply:
No, they didn't answer. We can disable this particular test for the moment, but I would like to keep it, because it fails very rare and at the same time helps to identify bugs if we change something in future.

Reply:
Yeah, so that's my concern: how would we notice that something went wrong if the test is marked as "it's ok for something to go wrong"?

Reply:
No way for the moment, my intention is to just keep this test somewhere and enable it when I do some related development. I can mark it with `skip`.

Reply:
ok, whatever you think is best

Reply:
Ok, I will mark it with `skip` so that it doesn't spam with all the xfail messages.